packages/tpetra/core/src/Tpetra_MultiVector_decl.hpp

```diff
@@ -841,6 +841,9 @@ namespace Tpetra {
     /// that one thread might win; it's that the value might get
     /// messed up.)
     ///
+    /// GPU synchronization semantics: Will sync if last access was on device.
+    /// Will not sync if last access was on host.
```

Comment:
Personally I don't think this particular one adds much value. There's already the section describing MV having DualView semantics before the class declaration starts which I think makes this behavior obvious.

That said, for things like `putScalar()` that have more complicated logic I could see this general kind of comment being useful.